### ISSUE TITLE
pdfmake silent fails if I add too many items

### ISSUE THREAD

**Issue by ragefuljoe** [bpampuch/pdfmake#246]
First, thanks for this great lib, it's so much better than what I was using.

I'm using pdfmake in my angularjs app. I generate png dataurls, and insert them with 

`dd.content.push({image:dataURL,width:550, style:'img'});`

If I try to add more than 3, nothing happens (I see the pdf dataurl in console but no print popup).

*edit*
the silent failing seems to only happen in chrome. I got things working ok in ff. safari is rendering tables off the page for some reason.

*edit2*
chrome download works, but open/print does not. can I just get the dataurl and use like jszip to save?

*update*
so it seems to be something up with chrome (42.0.2311.60). in playground, the examples for columns through images don't show anything. clicking open shows blank new window, print does nothing, but download works.
thanks!

**Comment by Kuzirashi** [comment]
This is really crucial bug. I developed my pdfmake component in Chrome, now it works on Firefox but prints grey page under Chrome v 42.0.2311.90 (64-bit). Removing the only 1 image from docDefinition solves problem. However this isn't a solution when I can't use images.

**Comment by jthoenes** [comment]
Can you provide me more details to reproduce the issue?

**Comment by Kuzirashi** [comment]
@jthoenes 
This is how I construct PDF:
~~~~ CoffeeScript
Ember.debug 'getPDF called'

      ###
       * Convert an image
       * to a base64 url
       * @param  {String}   url
       * @param  {Function} callback
       * @param  {String}   [outputFormat=image/png]
      ###
      convertImgToBase64URL = (url, callback, outputFormat) ->
        Ember.debug 'convertImgToBase64URL'
        canvas = document.createElement 'CANVAS'
        ctx = canvas.getContext '2d'
        img = new Image

        img.crossOrigin = 'Anonymous'
        img.onload = ->
          canvas.height = img.height
          canvas.width = img.width
          ctx.drawImage img, 0, 0
          dataURL = canvas.toDataURL outputFormat
          callback dataURL
          canvas = null

        img.src = url

      convertImgToBase64URL @get('model.imageBig'), (base64Img) =>
        Ember.debug 'Base64 convert callback called'

        docDefinition =
          content: [
            {
              columns: [
                {
                  text: """
                    #{@get 'model.name'}
                    #{@get 'model.time'}
                    #{@get 'calories'}/#{@get 'protein'}/#{@get 'carbo'}/#{@get 'fats'}
                    #{@get 'caloriesB'}/#{@get 'proteinB'}/#{@get 'carboB'}/#{@get 'fatsB'}
                  """
                }
                [
                  {
                    image: 'dishImage' # I have to comment this whole object out if I want PDF to be displayed in the browser instead of downloading it
                    width: 200
                    alignment: 'center'
                  }
                  {
                    text: "#{@get 'model.creditsName'}"
                    alignment: 'center'
                  }
                ]
              ]
            }
          ]
          images:
            dishImage: base64Img
          styles:
            sectionTitle:
              bold: true
              fontSize: 14
              margin: [0, 10, 0, 0]
            productsTable:
              margin: [0, 0, 0, 10]
            topMargin10:
              margin: [0, 10, 0, 0]

        pdfMake
        .createPdf docDefinition
        .open()

~~~~

What matters is only this part for me:
~~~~ CoffeeScript
 {
                    image: 'dishImage' # I have to comment this whole object out if I want PDF to be displayed in the browser instead of downloading it
                    width: 200
                    alignment: 'center'
                  }
~~~~

Changing `.open()` to `.download()` creates proper PDF but is downloaded instead of opened. It's not a big deal for me - download instead of open.

Also @ragefuljoe wrote that:
~~~~
In playground, the examples for columns through images don't show anything. clicking open shows blank new window, print does nothing, but download works.
~~~~

**Comment by jthoenes** [comment]
Hm, random thought: Maybe chrome reduced the maximum size for data urls?

We have large PDFs in our applications and at some time switched to store them in the temporary file system because the data urls got to big.

**Comment by jthoenes** [comment]
This is a problem with data urls. I created an issue for that, because other people also have experienced to it.

If you have any information on how to fix it, please add it to #275
